pandas/tests/frame/test_dtypes.py

```diff
@@ -724,6 +724,14 @@ def test_astype_duplicate_col(self):
         expected = concat([a1_str, b, a2_str], axis=1)
         tm.assert_frame_equal(result, expected)
 
+    def test_wrong_length_cat_dtype_raises(self):
+        # GH29523
+        cat = pd.Categorical.from_codes([0, 1, 1, 0, 1, 2], ["a", "b", "c"])
+        df = pd.DataFrame({"bar": range(10)})
+        err = "Length of values does not match length of index"
```

Comment:
can you move this test to pandas/tests/frame/test_indexing.py in the class TestDataFrameIndexingCategorical

Reply:
Done